app/tests/integration/nimbus/test_e2e_create_experiment.py

```diff
@@ -137,3 +137,113 @@ def test_create_new_experiment_remote_settings_timeout(
             break
     else:
         raise AssertionError("Timeout text was never shown.")
+
+
+def test_end_experiment_and_approve_end(
+    selenium, base_url, default_data, create_experiment, request, perform_kinto_action
+):
+    default_data.public_name = request.node.name
+
+    selenium.get(base_url)
+    home = HomePage(selenium, base_url).wait_for_page_to_load()
+    current_experiments = None
+    try:
+        current_experiments = len(home.tables[0].experiments)
+    except TimeoutException:
+        current_experiments = 0
+    review = create_experiment(selenium, home, default_data)
+    review.launch_without_preview.click()
+    review.request_review.click_launch_checkboxes()
+    review.request_review.request_launch_button.click()
+    review.approve()
```

Comment:
Wow this is so nice, great job on the refactor @lennartschoch and @jrbenny35 👏 👏 👏